index.rst

```diff
@@ -13,17 +13,188 @@
+At the same time, we have a need for service checkout and monitoring in a large number of deployments, many of which do not have access to large facility datasets (such as the hardware teststands). So system-test should focus on
+
+- single-service (to whatever degree practical) checkout and monitoring notebooks
+- ad-hoc troubleshooting notebooks (not run under mobu)
+
+In addition, we have identified a need for mobu-run notebooks that perform specific data checks, in some cases across deployments (for example: does the summit EFD and the USDF EFD agree on the number of measurements for a given topic during a particular time period).
+Given the strong semantic awareness of these checks and the need to compare data holdings across cluster, these will live in a separate repo (curation-check) and will contain payloads in separate directories that either (a) run on a specific deployment or (b) compare data in different data facilities.
+For the latter, given the fact that some of these comparisons will involve data holdings behind a VPN (such as the summit EFD, summit Butler etc), these should run from within a VPN deployment. Pressure of resources at the summit would suggest the best home for them would be to run under the base cluster's mobu.
+
+Service-specific checks
+-----------------------
+
+**Before:** Mobu can be configured to run all notebooks in a given directory and branch, but beyond that all notebook payloads in that directory/branch must work.
+
+**After:** By leveraging a naming convention that matches the application name in phalanx, mobu uses the ArgoCD UI or the environment manifest to only run notebooks corresponding to deployed services.
+
+**Discussion:** We want a more fine-grained way of running only notebooks that are expected to complete successfully.
+However an opt-in approach of creating manifests of specific notebooks in a repo (or a proxy of the same result, such as having mobu run off different branches in different environments) is a maintenance headache.
+If instead we adopt a naming convention for notebooks that match the ArgoCD application name, mobu can be modified to run all notebooks from participating services.
+For example, if at the La Serena base deployment nublado is set to true in the environment yaml and tap is set to false and there were notebooks in system-test named
+
+.. code-block:: bash
+
+      nublado_login.ipynb
+      nublado_dask_cluster.ipynb
+      tap.ipynb
+
+mobu would run the nublado notebooks but not attempt the tap one.
+
+The advantage of this approach is that developers can check in new notebooks for services without necessitating mobu changes.
+Note that currently mobu caches notebooks on start-up and needs to be restarted to pick up new notebooks; an API call or other method to induce a running mobu to re-poll its notebook repos would be useful (but not critical).
```

Comment:
We probably should just have mobu re-check its Git repository each time it finishes a cycle, unless you see a reason to want it to not switch notebooks until it's restarted.

Reply:
I worry that if we do as you say, when we are doing stress-testing/characterisation with mobu being the face that launches a thousand users, github will get mad at us (cause I assume the refresh would be per mobu user not per mobu instance but maybe I am totally wrong here - am I?

Reply:
Good point, rather than switching at the end of each cycle we'd want to do this as a background task that periodically checks if there's a newer repository available and if so checks it out so that the runners can switch to the new one at the end of their next cycle.

Reply:
On discussion we agreed that the correct solution for everyone's concerns is to 
(a) register a webhook with Github so you push the command to mobu's API to refresh a newly merged notebook
(b) always re-reread on mobu restart so that sites with no inbound internet (eg summit) would still work

Reply:
Added a section capturing this discussion:
 https://github.com/lsst-sqre/sqr-080/commit/39d2f84ce5b49775094a696e60ae456a06860b68?diff=unified&w=0